### PR TITLE
Staging v1.8.0 beta.1

### LINKED ISSUE #889: [Release] [1.8.0-beta.1] Release tasks
## Release Requirements Checklist for a Beta
This checklist is a chronological ordering of the release tasks to complete for an IBM Ansible z/OS core collection. This template is for a Beta release that will release to Galaxy as well. If you are following this issue, note that the version number is subject to change based on a successful release to Galaxy. 

Please check off each task as it completes.

## General workflow
This a subsequent release (not a patch) thus these releases tasks will eventually be merged into`main` and into `dev` to allow for the meta data updates to propagate as well as the release tag. 

### (0) Pre-release tasks
- [x] Create a staging branch where code is considered frozen and only metadata related commits will be performed, for example, updating galaxy.yml, runtime.yml, generating module doc if needed, release notes etc.
    - [x] `git checkout dev`
    - [x] `git pull`
    - [x] `git checkout -b staging-v<version>`
    - [x] `git push -u origin staging-v<version>`
- [x] Inform the team the `dev` branch is now open for next release contributions

### (1) Technical Writer Tasks
- [x] Ensure all technical writer Git issues are resolved for this release

### (2) Release Tasks for `release-tasks-v<version>` branch:
- [x] Update `galaxy.yml`
- [x] Update `meta/runtime.yml` - align the ansible-core version to the [AAP life cycle](https://access.redhat.com/support/policy/updates/ansible-automation-platform) or check with the team.
- [x] Update `meta/ibm_zos_core_meta.yml`
- [x] Update `README.md` - particularly review [Ansible version compatibility](https://github.com/ansible-collections/ibm_zos_core#ansible-version-compatibility) and [IBM z/OS core collection](https://github.com/ansible-collections/ibm_zos_core#ibm-zos-core-collection) sections.
- [x]  Update `.ansible-lint` config such that `galaxy.yml` and `build_ignore` entries are in sync.
- [x] Review copyrights are updated as needed, e.g. current year if changes are made. Copyright years contain 2 years, a start year when the source was created followed by a comma then the last year it was modified, eg `2020, 2023`.
- [x] Generate module docs (restructured text - RST) and check it into the release branch
- [x] Add a change log summary fragment in the `/changelog` directory for this release:
       - Create a file called "v\<major\>.\<minor\>.\<patch\>_summary.yml", eg `v1.6.0_summary.yml`
       - Place it under `ibm_zos_core/changelogs/fragments/`
       - Add contents (update the date, don't go past the current day):
```
release_summary: |
  Release Date: '2023-06-23'
  This changelog describes all changes made to the modules and plugins included
  in this collection. The release date is the date the changelog is created.
  For additional details such as required dependencies and availability review
  the collections `release notes <https://ibm.github.io/z_ansible_collections_doc/ibm_zos_core/docs/source/release_notes.html>`__
```
- [x] Generate CHANGELOG.rst
    - `antsibull-changelog lint`
    - `antsibull-changelog release` or `antsibull-changelog release -v`
- [x] Create release notes (`docs/source/release_notes.rst`)
- [x] Review all module source imports; ensure sure no new imports are subject to license discrepancies 

### (3) Quality assurance
- [x] Run a Mend scan and check the results into the designated repository
- [x] Run a Bandit scan and check the results into the designated repository  
- [x] Ansible-lint validation , manual step at the moment, choose from one of the below to run ansible-lint:
    - ./ac --ac-lint
    - ./ac --venv-start
       - ansible-lint -c .ansible-lint --profile production
- [x] Sanity tests, use the pipeline option and choose to run only the sanity tests by checking `ANSIBLE_TEST_ONLY` which specifies only Ansible Certification tests and security scan run, meaning no functional tests run." As a backup option, you can use the `./ac` tool with similar options `./ac --ac-sanity --version 3.10` to run sanity.
- [x] Run the galaxy importer (currently a manual step as its not part of the pipeline yet)
   - [x] `python -m galaxy_importer.main ibm-ibm_zos_core-<version>.tar.gz`. You can perform this with commands:
        ```
        $ ./ac --venv-start
        $ python -m galaxy_importer.main ibm-ibm_zos_core-1.7.0.tar.gz
        ```
   - [x] Check output log for any errors/issues. For success, look for the following msg:
        ```
         Collection loading complete
         Importer processing completed successfully
        ```
- [x] Full pipeline regression at 100% success on either `release-tasks-v<version>` or `stagging-v<version>`:
     - [x] Certify on IBM Enterprise Python versions 3.11  or latest (with the latest PTF for env var PYTHONSTDINENCODING)
     - [x] Certify on IBM Enterprise Python and new environment var PYTHONSTDINENCODING
     - [x] Certify on ZOAU 1.24
     - [x] Certify on `ansible-core v2.15.x` or latest supported by Ansible Automation Platform
     - [x] Ensure `RUN_ALL_TESTS` is selected

### (5) Validation
- [x] Build the image from `staging-v<version>` branch, unpack it up, inspect the contents. Ensure only the required files are contained in the package. 
- [x] Test collection quality with an upload to the IBM HCF development Galaxy Server by uploading the build from branch `staging-v<version>`. This first build test is to alert you to any issues that might need to be made before opening a pull request.

### (6) GitHub
- [x] Open a pull request for `staging-v<version>` against `main`.
- [x] Create a [draft release](https://github.com/ansible-collections/ibm_zos_core/releases/new) in GitHub, for:
    - "Choose a tag", enter `v<version>`, eg `v1.6.0`
    - "Target" , enter `main`
    - "Release title" , enter `release-v<version>`, eg `release-v1.6.0`
    - "Describe this release", enter the contents created for `doc/source/release_notes.rst`, you may need to remove trailing `_` from the RST you paste since the description is supposed to be Markdown.
- [x] When the pull request has been approved, do not merge it, build the collection again and test the collection quality again with a new build from the reviewed `staging-v<version>` branch by uploading it to the Ansible development Galaxy Server. (Do not merge the pull request till after AAP and Galaxy success.)

### (7) Release tasks (**In this order**)
- [ ] ~~Upload to [AAP](https://console.redhat.com/ansible/automation-hub), on success continue to the next step.~~
- [x] Upload to [Galaxy](https://galaxy.ansible.com/ibm/ibm_zos_core), on success continue to the next step.
- [x] Merge pull request for `staging-v<version>` against `main` and **DO NOT** delete the `staging-v<version>` branch, you will need the `staging-v<version>` branch soon enough.
- [x] Publish the Github release draft, include the collection archive as part of the release.
- [x] Open a Git issue in [z_ansible_collections_doc](https://github.com/IBM/z_ansible_collections_doc/issues) to have [Red Hat® Ansible Certified Content for IBM Z docs](https://ibm.github.io/z_ansible_collections_doc/index.html) generated
- [x] Push `stagging-v<version>` changes into `dev` to ensure any fixes and meta are propagated upstream. There are different ways to do this, I find a merge ends up complicating things because `dev` will have received an entire quarters of changes and now old changes are pushed upstream. Generally, what is when performing the release tasks from bullet (2), I commit each as a separate so that I can cherry pick them later.All that should change is metadata , eg `galaxy.yml` , `release_notes.rst`, etc, no source. In case there is a source change, separate commits makes it easy to cherry pick. Optionally (usually what i do),  you can cherry pick the PR which will have by default squashed all the individual commits. For example, if you want to cherry-pick the squashed PR, get the commit id from the PR then perform the following commands:
    ```
    git checkout dev <-- check out dev since this is where we want to cherry-pick to
    git pull                 <-- pull the latest
    git checkout -b cherry-picked-170-into-dev <-- create a new branch
    git cherry-pick 472611  <-- cherry-pick the squashed PR 
    git push -u origin cherry-picked-170-into-dev <-- push the changes and then open a PR against dev
    ```

### (8) Log collection
- [x] Copy any additional playbook or manual tests performed into the release folder
- [x] Copy the pipeline log from Jenkins into the release folder
- [x] Copy the Automation Hub import log into the release folder
- [x] Copy the Galaxy import log in the release folder
- [ ] Copy the git log (`git log --pretty=oneline`) into the release folder

### PR DESCRIPTION
**Pull request by fernandofloresg**
##### SUMMARY

Staging v1.8.0-beta.1 branch to merge into main

Fixes #889 

##### ISSUE TYPE

- Release Pull Request

### CODE REVIEW

File: scripts/mounts.env [hunk 1]
```diff
@@ -56,7 +58,8 @@ python_mount_list_str="1:3.8.2:/allpython/3.8.2/usr/lpp/IBM/cyp/v3r8/pyz:/allpyt
 "3:3.9:/allpython/3.9/usr/lpp/IBM/cyp/v3r9/pyz:/allpython/3.9:IMSTESTU.PYZ.V39016.ZFS "\
 "4:3.10:/allpython/3.10/usr/lpp/IBM/cyp/v3r10/pyz:/allpython/3.10:IMSTESTU.PYZ.V3A09.ZFS "\
 "5:3.11:/allpython/3.11/usr/lpp/IBM/cyp/v3r11/pyz:/allpython/3.11:IMSTESTU.PYZ.V3B02.ZFS "\
```
Comment (richp405) [inline scripts/mounts.env:60]: technically, this (5) has been changed from 3.11 to 3.11-5

File: plugins/action/zos_copy.py [hunk 2]
```diff
@@ -59,6 +59,12 @@ def run(self, tmp=None, task_vars=None):
         local_follow = _process_boolean(task_args.get('local_follow'), default=False)
         remote_src = _process_boolean(task_args.get('remote_src'), default=False)
         is_binary = _process_boolean(task_args.get('is_binary'), default=False)
+        force_lock = _process_boolean(task_args.get('force_lock'), default=False)
+<<<<<<< HEAD
```
Comment (richp405) [inline plugins/action/zos_copy.py]: I don't think this merge text should be here.

**Review by richp405 — CHANGES_REQUESTED** [review]
I found merge artifact in source.

**Review by richp405 — APPROVED** [review]

**Review by rexemin — APPROVED** [review]
As far as I can tell, it has everything it should go into 1.8.0-beta.1, thank you so much for taking care of this release!

**Review by AndreMarcel99 — APPROVED** [review]

File: scripts/mounts.env (hunk 1 above)
Comment (ddimatos) [reply scripts/mounts.env:60]: I'll update the `.ac` and other files as part of the work load balancer work I have for next sprint, I think all the config files for .ac will need refreshing and new ones for ansible-core.2.16 so i can all be handled there.

**Review by ddimatos — APPROVED** [review]
Looks good to me, thanks for this @fernandofloresg , nice work.

**Pull request merged**